### DISCUSSION TITLE
Help :  ERROR:flask_appbuilder.api:'SupersetApp' object has no attribute 'json_encoder'

### DISCUSSION THREAD

**Discussion by Attachai77** [apache/superset#28138] [Q&A / Help]
How to reproduce it
- git clone https://github.com/apache/superset.git
- added `SUPERSET_CONFIG_PATH=/app/docker/pythonpath_dev/superset_config.py` into .env
- docker compose -f docker-compose-non-dev.yml up -d
- send request login with postman
- got error as image below.

<img width="637" alt="Screenshot 2567-04-19 at 10 14 56" src="https://github.com/apache/superset/assets/15119618/2f0a8523-cbff-4900-9886-976537e33fc9">

<img width="1177" alt="Screenshot 2567-04-19 at 10 24 51" src="https://github.com/apache/superset/assets/15119618/a6e25af3-4f70-4b7b-8758-500e51688e1e">

**Comment by ducnt7-rks** [comment]
I encountered this problem same you and I tried pip install json-encoder but not works

**Comment by teeppiphatarv** [comment]
I encountered the same problem as you. Now fixed by using version 3.1.2

**Comment by marrocksd** [comment]
For those who is having this issue. The latest versions of python have built-in json_encoder method, so latest version of Flask uses that instead of their custom one, and they actually removed that from Flask.

Not sure but I think it's a problem from Flask-AppBuilder which is OOS here.
If you guys want to use version 4.x, downgrade to Flask==2.2.5 by adding it to docker/requirements-local.txt

**Reply by vtttuong** [reply]
thank you, i fixed
